File: 考研计算机/数据结构/code/ds/_8_algo/练习/对撞指针技巧及其代码模版/CollisionPointer.java
package com.laotang.第_1_章_数组和字符串.对撞指针技巧及其代码模版;

public class CollisionPointer {

    // 数组反转
    public void reverse(int[] nums) {
        int left = 0, right = nums.length - 1;
        while (left < right) {
            int tmp = nums[left];
            nums[left] = nums[right];
            nums[right] = tmp;

            left++;
            right--;
        }
    }
}

File: 考研计算机/数据结构/code/ds/_8_algo/练习/对撞指针技巧及其代码模版/CollisionPointer.go
// 数组反转
func reverse(nums []int) {
    left, right := 0, len(nums) - 1
    for left < right {
        tmp := nums[left]
        nums[left] = nums[right]
        nums[right] = tmp

        left++
        right--
    }
}

// 验证回文串
func isPalindrome(s string) bool {
    left, right := 0, len(s) - 1
    for left < right {
        if s[left] != s[right] {
            return false
        }
        left++
        right--
    }
    return true
}